subsys/caf/modules/sensor_data_aggregator.c

```diff
@@ -138,26 +142,26 @@ static void send_buffer(struct aggregator *agg, struct aggregator_buffer *ab)
 
 static int enqueue_sample(struct aggregator *agg, struct sensor_event *event)
 {
-	if (event->dyndata.size != agg->sensor_data_size) {
+	if ((event->dyndata.size) != ((agg->values_in_sample) * (sizeof(struct sensor_value)))) {
 		return -EBADMSG;
 	}
 	if (!agg->active_buf) {
 		return -ENOMEM;
 	}
 	struct aggregator_buffer *ab = agg->active_buf;
 
-	size_t pos = ab->sample_cnt * agg->sensor_data_size;
-	size_t avail = agg->buf_len - pos;
+	size_t pos = ab->sample_cnt * agg->values_in_sample;
+	size_t avail = agg->buf_len - pos*sizeof(struct sensor_value);
 
-	if (avail < agg->sensor_data_size) {
+	if (avail < agg->values_in_sample) {
```

Comment:
This is wrong. `avail` means bytes available and you are comparing this to numer of `sensor_values` in single `sample`. It should be:
``` suggestion
	if (avail < (agg->values_in_sample * sizeof(struct sensor_value))) {
```

Generally i don'y quite like lines L145-L156, here is why:
You are changing `agg->sensor_data_size` to `agg->values_in_sample * sizeof(struct sensor_value)` in line 145,
In line L153 you are chaning same `agg->sensor_data_size` to `agg->values_in_sample` to correctly calculate `pos`.
Speaking of that, `pos` has change its meaning.

Discussed offline: There will be a small refactor to this part as it introduces a bit of confusion when referring to the buffer as byte/non byte buffer.